src/ui/public/agg_types/buckets/date_histogram.js

```diff
@@ -156,7 +158,42 @@ export function AggTypesBucketsDateHistogramProvider(timefilter, config, Private
             return;
           }
         }
+      },
+
+      {
+        name: 'isFilteredByKbnTime',
+        default: true,
+        write: _.noop
       }
-    ]
+    ],
+    getRequestAggs: function (agg) {
+      const aggs = [];
+
+      // Filter only needed when date field is not index pattern time field
+      if (!agg.fieldIsTimeField() && agg.params.isFilteredByKbnTime && agg.getField()) {
```

Comment:
That's really unfortunate. I'm not familiar with the visualization code, but ideally there's a way to get the query to add an extra time filter. I'll remove my blockage.